### ISSUE TITLE
Question: Using the priority for module:migrate?

### ISSUE THREAD

**Issue by voydz** [pingpong-labs/modules#116]
Hey there,

we use the module component in a new project. But recently we stumbled upon some problems with migration and seeding.

**Our situation**

We are having some modules which are acutally dependent on others. In some migrations we are setting up foreign keys and things like that. Those migrations are going horribly wrong since most of the Commands use the `Repository::all()` method (no ordering is going to happen). 

**My Thoughts**

If `Repository::all()` is used even there is a sorting mechanic already, there has to be a reason. I am curious and I want to know why. :)

To resolve the issue we have thought of two different approaches:

1. Module inheritance as stated in #45 (commands should reflect the inheritance and i.e. migrate up the dependency tree)
2. Using the priority flag with `Repository::getOrdered()`. (But then, the next thing is `priority` itself does not work quite well IMHO.)

**The problem I see with priority**

It is about the bounds and range of different modules. To make a module the most important, you have to define a really high value and you generally have no idea what the upper bounds could be, because it depends on how much modules will be installed. Is there even a use case for the `priority` property if not in my own modules?

Now what about this? Just flip it. Use a simple `order` approach. The most important module is 0. It is well defined lower bound (you do not have to think about) and each module following has its `order` increased by 1.

I just want to hear your opinions on that topic. I could think of making a pull request if you agree with one of those.

**Comment by fgreinus** [comment]
:+1:

**Comment by voydz** [comment]
Yeah @fgreinus you are right. The implementation of those solutions are pretty simple to make at least migrations working.

**Comment by gravitano** [comment]
Hi @voydz, 

I agree with you. I will make changes to this issue.

Thank you.

**Comment by voydz** [comment]
@gravitano awesome, thank you!
